### PR TITLE
[Merged by Bors] - chore: a batch of whitespace fixes

### PR DESCRIPTION
**Pull request by adomani**
Found by #26926.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Integral/TorusIntegral.lean [hunk 1]
```diff
@@ -140,7 +140,7 @@ def torusIntegral (f : ℂⁿ → E) (c : ℂⁿ) (R : ℝⁿ) :=
   ∫ θ : ℝⁿ in Icc (0 : ℝⁿ) fun _ => 2 * π, (∏ i, R i * exp (θ i * I) * I : ℂ) • f (torusMap c R θ)
 
 @[inherit_doc torusIntegral]
-notation3"∯ "(...)" in ""T("c", "R")"", "r:(scoped f => torusIntegral f c R) => r
+notation3 "∯ " (...) " in " "T(" c ", " R ")" ", " r:(scoped f => torusIntegral f c R) => r
```
Comment (adomani) [inline Mathlib/MeasureTheory/Integral/TorusIntegral.lean:143]: No pretty-printing implication, just preferred formatting.

File: Mathlib/Algebra/BigOperators/Finprod.lean [hunk 2]
```diff
@@ -112,12 +112,12 @@ open Batteries.ExtendedBinder
 /-- `∑ᶠ x, f x` is notation for `finsum f`. It is the sum of `f x`, where `x` ranges over the
 support of `f`, if it's finite, zero otherwise. Taking the sum over multiple arguments or
 conditions is possible, e.g. `∏ᶠ (x) (y), f x y` and `∏ᶠ (x) (h: x ∈ s), f x` -/
-notation3"∑ᶠ "(...)", "r:67:(scoped f => finsum f) => r
+notation3"∑ᶠ " (...) ", " r:67:(scoped f => finsum f) => r
```
Comment (adomani) [inline Mathlib/Algebra/BigOperators/Finprod.lean:115]: No pretty-printing implication in this file, just preferred formatting.

File: Mathlib/MeasureTheory/Measure/FiniteMeasure.lean [hunk 3]
```diff
@@ -405,7 +405,7 @@ theorem testAgainstNN_lipschitz_estimate (μ : FiniteMeasure Ω) (f g : Ω →
   intro ω
   have le' : f ω ≤ g ω + nndist f g := by
     calc f ω
-     _ ≤ g ω + nndist (f ω) (g ω)     := NNReal.le_add_nndist (f ω) (g ω)
+     _ ≤ g ω + nndist (f ω) (g ω) := NNReal.le_add_nndist (f ω) (g ω)
      _ ≤ g ω + nndist f g             := (add_le_add_iff_left (g ω)).mpr (le_dist ω)
```
Comment (JovanGerb) [inline Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: This seems suspicious. Why was only one of the whitespaces removed?

Comment (adomani) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: Good catch!  This is one of those rare cases in which the pretty-printer does more than adjust whitespace, but also *adds* parentheses.  It would want `NNReal.le_add_nndist (f ω) (g ω)` to be `(NNReal.le_add_nndist (f ω) (g ω))` instead.

However, the linter currently assumes that the only discrepancies between the pretty-printed syntax and the input are only whitespace, gets confused and stops processing the rest of the command.

I will remove the spaces for consistencies, but I am tempted to add a TODO for the parentheses issue, rather than fixing it right now.

Comment (adomani) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: Btw, if I add parentheses where the pretty-printer puts them, then the following whitespaces are highlighted.

Comment (JovanGerb) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: Surely, we don't want to put brackets there though? Just like we don't have brackets around `(add_le_add_iff_left (g ω)).mpr (le_dist ω)`. I would consider this to be a mistake by the pretty printer.

Comment (JovanGerb) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: Although there are more pretty printing issues with `calc`. Like do we want
```
calc X
  _ = Y := sorry
  _ = Z := sorry
```
or
```
calc X
    = Y := sorry
  _ = Z := sorry
```
or
```
calc X = Y := sorry
  _ = Z := sorry
```

Comment (adomani) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: Ah, I added the parentheses to confirm that the linter would flag the rest, but then I forgot to remove them.  Let me do that!

As for the other versions, the linter allows a few options, but I forget exactly which ones.

Comment (adomani) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: In your `calc` examples, is the first actually allowed, with both `X` and `_`?

Comment (JovanGerb) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: Apparently it is allowed, because that is what this `calc` is doing that we are looking at.

Comment (adomani) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: Wow, you are right!  I did not know that this was a possibility.  Well, it seems that the linter allows it as well!  :smile:

Comment (grunweg) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: (Can you add tests for these to the linter? It's really hard to keep track of all these edge cases otherwise.)

File: Mathlib/RingTheory/Flat/EquationalCriterion.lean [hunk 4]
```diff
@@ -130,11 +130,10 @@ theorem tfae_equational_criterion : List.TFAE [
     let x' : Fin l → M := fun i ↦ x (single i 1)
     have := calc
       ∑ i, f' i • x' i
-      _ = ∑ i, f i • x (single i 1)         := rfl
+      _ = ∑ i, f i • x (single i 1) := rfl
       _ = x (∑ i, f i • Finsupp.single i 1) := by simp_rw [map_sum, map_smul]
-      _ = x f                               := by
-        simp_rw [smul_single, smul_eq_mul, mul_one, univ_sum_single]
-      _ = 0                                 := hfx
+      _ = x f := by simp_rw [smul_single, smul_eq_mul, mul_one, univ_sum_single]
```
Comment (grunweg) [inline Mathlib/RingTheory/Flat/EquationalCriterion.lean:135]: Much better, thanks!

**Review by grunweg — COMMENTED** [review]
Thanks!
maintainer merge

File: Mathlib/MeasureTheory/Measure/FiniteMeasure.lean (hunk 3 above)
Comment (adomani) [reply Mathlib/MeasureTheory/Measure/FiniteMeasure.lean]: I will, but in the PR where the linter actually has the issue: here the linter is not inspecting this part of the declaration.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge
